tests/api/api/test_grafana_proxy.py

```diff
@@ -0,0 +1,170 @@
+    )
+    assert response.status_code == 200
+
+
+@pytest.mark.skipif(
+    _is_env_params_dont_exist(), reason=_build_skip_message(),
+)
+def test_grafana_list_endpoints(db: Session, client: TestClient):
+    endpoints_in = [_mock_random_endpoint("active") for _ in range(5)]
+
+    for endpoint in endpoints_in:
+        _write_endpoint_to_kv(endpoint)
+
+    response = client.post(
+        url="/api/grafana-proxy/model-endpoints/query",
+        headers={"X-V3io-Session-Key": _get_access_key()},
+        json={"targets": [{"target": "project=test;target_endpoint=list_endpoints"}]},
+    )
+
+    response_json = response.json()
+    if not response_json:
+        fail(f"Empty response, expected list of dictionaries. {response_json}")
+
+    response_json = response_json[0]
+    if not response_json:
+        fail(
+            f"Empty dictionary, expected dictionary with 'columns', 'rows' and 'type' fields. {response_json}"
+        )
+
+    if "columns" not in response_json:
+        fail(f"Missing 'columns' key in response dictionary. {response_json}")
+
+    if "rows" not in response_json:
+        fail(f"Missing 'rows' key in response dictionary. {response_json}")
+
+    if "type" not in response_json:
+        fail(f"Missing 'type' key in response dictionary. {response_json}")
+
+    assert len(response_json["rows"]) == 5
+
+
+def test_parse_query_parameters_should_fail():
+    # No 'targets' in body
+    with pytest.raises(MLRunBadRequestError):
+        _parse_query_parameters({})
+
+    # No 'target' list in 'targets' dictionary
+    with pytest.raises(MLRunBadRequestError):
+        _parse_query_parameters({"targets": []})
+
+    # Target query not separated by equals ('=') char
+    with pytest.raises(MLRunBadRequestError):
+        _parse_query_parameters({"targets": [{"target": "test"}]})
+
+
+def test_parse_query_parameters_should_not_fail():
+    # Target query separated by equals ('=') char
+    params = _parse_query_parameters({"targets": [{"target": "test=some_test"}]})
+    assert params["test"] == "some_test"
+
+    # Target query separated by equals ('=') char (multiple queries)
+    params = _parse_query_parameters(
+        {"targets": [{"target": "test=some_test;another_test=some_other_test"}]}
+    )
+    assert params["test"] == "some_test"
+    assert params["another_test"] == "some_other_test"
+
+
+def test_validate_query_parameters_should_fail():
+    # No 'target_endpoint' in query parameters
+    with pytest.raises(MLRunBadRequestError):
+        _validate_query_parameters({})
+
+    # target_endpoint unsupported
+    with pytest.raises(MLRunBadRequestError):
+        _validate_query_parameters({"target_endpoint": "unsupported_endpoint"})
+
+
+def test_validate_query_parameters_should_not_fail():
+    _validate_query_parameters({"target_endpoint": "list_endpoints"})
+
+
+def _get_access_key() -> Optional[str]:
+    return os.environ.get("V3IO_ACCESS_KEY")
+
+
+@pytest.fixture(autouse=True)
```

Comment:
When you add it here, does it affect all tests everywhere ? or only the tests in this file ?
If the first, then it can't be kept with `autouse=True`

Reply:
When you say rest of tests, what rest are you talking about?

Reply:
rest is any test not in this file

Reply:
No, it does not apply to the rest of the tests